pkg/engine/cas_template_engine.go

```diff
@@ -138,6 +138,46 @@ func NewCASEngine(casTemplate *v1alpha1.CASTemplate, runtimeKey string, runtimeV
 	return
 }
 
+// addConfigToConfigTLP will add final cas volume configurations to ConfigTLP.
+//
+// NOTE:
+//  This will enable templating a run task template as follows:
+//
+// {{ .Config.<ConfigName>.enabled }}
+// {{ .Config.<ConfigName>.value }}
+//
+// NOTE:
+//  Above parsing scheme is translated by running `go template` against the run
+// task template
+func (c *CASEngine) addConfigToConfigTLP() error {
+	var configName string
+	allConfigsHierarchy := map[string]interface{}{}
+	allConfigs := c.casTemplate.Spec.Defaults
+
+	for _, config := range allConfigs {
+		configName = strings.TrimSpace(config.Name)
+		if len(configName) == 0 {
+			return fmt.Errorf("failed to merge config '%#v': missing config name", config)
```

Comment:
error message need not have 'merge'. Can below be used?
`fmt.Errorf("failed to add config as a top level property: missing config name: config '%+v'", config)`